### PR TITLE
Mitigate intermittent test_buckets and shuffle_smoke_test OOM issue [databricks]

### PR DESCRIPTION
**Pull request by pxLi**
related to [#10940](https://github.com/NVIDIA/spark-rapids/issues/10940)

This is not a fix to #10940. 
1. Try increase the default `driver.memory` (1g->1.5g) in the test script to mitigate the intermittent `SparkOutOfMemoryError` while test_buckets.
2. disable xdist for shuffle_smoke_test in databricks (now be consistent with non-databricks)  to avoid intermittent OOM kill (previously also intermittently appears in non-related CI)

this would help local mode run. Standalone run we overwrote the executor.memory in the nightly, so we have not seen any repro in https://github.com/NVIDIA/spark-rapids/blob/branch-24.08/jenkins/spark-tests.sh#L165-L189

### CODE REVIEW

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

File: integration_tests/run_pyspark_from_build.sh [hunk 1]
```diff
@@ -268,6 +268,10 @@ else
         export PYSP_TEST_spark_driver_memory="${DRIVER_MEMORY}"
     fi
 
+    # TODO: https://github.com/NVIDIA/spark-rapids/issues/10940
+    EXECUTOR_MEMORY=${EXECUTOR_MEMORY:-"1536m"}
+    export PYSP_TEST_spark_executor_memory=${PYSP_TEST_spark_executor_memory:-"${EXECUTOR_MEMORY}"}
```
Comment (gerashegalov) [inline integration_tests/run_pyspark_from_build.sh:273]: Most of the tests run  Spark in the local mode. If spark master is local we need spark.driver.memory

Comment (pxLi) [reply integration_tests/run_pyspark_from_build.sh:273]: I think for local mode, spark would combine driver+executor memory request, no?

Comment (pxLi) [reply integration_tests/run_pyspark_from_build.sh:273]: Did a quick verification in spark320. I was wrong, executor memory was ignored in local mode at least for spark 320. let me update the code, thanks

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
crashed java agent with the 1.5g memory in our databricks CI (OOM kill)
```
============ 55 failed, 30 passed, 36 warnings in 78.79s (0:01:18) =============
```

let me poke a little bit more

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
run a few more times

**Comment by pxLi** [comment]
again sonatype

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
csp capacity failure

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
shuffle test failed oom kill again.

I will try limit shuffle cases later

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

File: integration_tests/run_pyspark_from_build.sh [hunk 2]
```diff
@@ -129,7 +129,7 @@ else
         # below where the processes are launched.
         GPU_MEM_PARALLEL=`nvidia-smi --query-gpu=memory.free --format=csv,noheader | awk '{if (MAX < $1){ MAX = $1}} END {print int((MAX - 2 * 1024) / ((1.5 * 1024) + 750))}'`
         CPU_CORES=`nproc`
-        HOST_MEM_PARALLEL=`cat /proc/meminfo | grep MemAvailable | awk '{print int($2 / (5 * 1024 * 1024))}'`
+        HOST_MEM_PARALLEL=`cat /proc/meminfo | grep MemAvailable | awk '{print int($2 / (5 * 1024 * 1536))}'`
```
Comment (revans2) [inline integration_tests/run_pyspark_from_build.sh:132]: Can you please update the comments just above this as the amount is no longer 5 GiB?  Also the `5 * 1024 * 1024` was clear what it was, but `5 * 1025 * 1536` is much less clear that it is 7.5 GiB.  Could we try and rearrange that to make it more clear? `7.5 * 1024 * 1024` perhaps?

**Review by revans2 — APPROVED** [review]
Mostly nits about making the code/comments consistent and more readable.

File: integration_tests/run_pyspark_from_build.sh (hunk 2 above)
Comment (pxLi) [reply integration_tests/run_pyspark_from_build.sh:132]: sure, thanks for the suggestion

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Comment by pxLi** [comment]
build

**Review by NvTimLiu — APPROVED** [review]
LGTM

**Review by yinqingh — APPROVED** [review]

**Pull request merged**